source/agora/test/Base.d

```diff
@@ -1958,6 +1996,27 @@ public struct TestConf
     /// Transaction put into the relay queue will expire, and will be removed
     /// after `relay_tx_cache_exp`.
     Duration relay_tx_cache_exp = 60.minutes;
+
+    /// true, if this node should collect statistics about other
+    /// nodes in the network, including their geographical location and OS
+    public bool collect_network_statistics = false;
+
+    /// The number of network crawlers that will be instantiated to collects
+    /// statistics about other nodes in the network
+    public ubyte num_of_crawlers = 1;
+
+    /// The number of seconds one crawler should wait after successfully contacted
+    /// a node
+    public Duration crawling_interval = 2.seconds;
+
+    /// true, if NetworkManager.getClient should return null, as opposed to assert,
+    /// in case the node passed to getClient has already been shut down or
+    /// never created
+    public bool use_non_assert_get_client = false;
+
+    /// true, if NetworkManager should automatically initiate network discovery,
+    /// when the node is started up
+    public bool do_network_discovery = true;
```

Comment:
I'm worried that so much of the crawler internal ends up being booleans in the global unittest configuration.

Reply:
then what is your suggestion?